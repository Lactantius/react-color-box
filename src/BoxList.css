.BoxList-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 10px;
  margin: 2rem;
  justify-items: center;
  align-items: center;
}
