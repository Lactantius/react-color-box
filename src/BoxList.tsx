import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Box, { BoxStyle } from "./Box";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css";

function BoxList(): JSX.Element {
  const [boxen, setBoxen] = useState([
    { width: "20px", height: "60px", color: "#ff0000", id: uuid() },
  ]);

  const addBox = (box: BoxStyle) =>
    setBoxen((oldBoxen) => [...oldBoxen, { ...box, id: uuid() }]);

  const removeBox = (id: string) =>
    setBoxen((oldBoxen) => oldBoxen.filter((box) => id !== box.id));

  return (
    <div className="BoxList">
      <NewBoxForm add={addBox} />
      <div className="BoxList-container">
        {boxen.map((box) => (
          <Box
            key={box.id}
            width={box.width}
            height={box.height}
            color={box.color}
            id={box.id}
            remove={removeBox}
          />
        ))}
      </div>
    </div>
  );
}

export default BoxList;
