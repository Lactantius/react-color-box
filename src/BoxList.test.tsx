import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

jest.mock("uuid", () => ({ v4: () => "123456789" }));

it("renders", () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
