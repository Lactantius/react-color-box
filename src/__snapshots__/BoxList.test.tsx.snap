// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm"
    >
      <label
        for="width"
      >
        Box Width
      </label>
      <input
        name="width"
        placeholder="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Box Height
      </label>
      <input
        name="height"
        placeholder="height"
        type="number"
        value="0"
      />
      <label
        for="color"
      >
        Box Color
      </label>
      <input
        class="NewBoxForm-color-picker"
        name="color"
        placeholder="color"
        type="color"
        value="#000000"
      />
      <button>
        New Box
      </button>
    </form>
    <div
      class="BoxList-container"
    >
      <div
        class="Box"
        id="123456789"
        style="width: 20px; height: 60px; background-color: rgb(255, 0, 0);"
      >
        <button
          class="Box-remove-button"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
